src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/resource/AuditResource.kt

```diff
@@ -87,9 +92,46 @@ class AuditResource(
     @RequestParam who: String? = null,
     @RequestParam what: String? = null,
   ): Page<AuditDto> = auditService.findPage(pageable, who, what)
+
+  @PreAuthorize("hasRole('ROLE_AUDIT')")
+  @PostMapping("")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Add a new audit event",
+    description = "Adds a new Audit Event to the audit queue, role required is ROLE_AUDIT",
+    security = [SecurityRequirement(name = "ROLE_AUDIT")],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [Content(mediaType = "application/json")]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Audit Event Added to audit event queue",
+        content = [Content(mediaType = "application/json")]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Invalid request to add an audit event",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint, requires a valid OAuth2 token",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an authorisation with role ROLE_AUDIT",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  fun insertAuditEvent(@RequestBody auditEvent: AuditEvent) {
+    auditService.sendAuditEvent(auditEvent)
```

Comment:
@karenmillermoj when this is all working I think we can do some nice stuff around operationId, we should be able to grab it from the request using the Telemetry client and pass it with the message, no need for most clients to explicitly set it since it will end up in the header automatically. But that is future work